src/contexts/ModalContext.tsx

```diff
@@ -0,0 +1,38 @@
+'use client'
+
+import { PropsWithChildren, createContext, useContext, useState } from 'react'
+
+export type ModalContextType = {
+  profileId: string
+  setProfileId: (id: string) => void
+  showRejectModal: boolean
```

Comment:
**nitpick:** I think we should only have (of course we we assume that only one modal can be visible at the time, but let's leave it for now:

```ts
export type ModalContextType = {
  isModalVisible: boolean;
  showModal: (state: boolean) => void
}
```